terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+  vpc_id      = module.vpc.vpc_id
+
+  # cassandra non-TLS
+  egress {
+    from_port   = 7000
+    to_port     = 7000
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # cassandra TLS
+  egress {
+    from_port   = 9160
+    to_port     = 9160
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # ElasticSearch
+  egress {
+    from_port   = 9300
+    to_port     = 9300
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # minio
+  egress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # minio
+  egress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }  
+
+  tags = {
+    Name = "ephemeral_private"
+  }
+}
+
+# A security group for ephemeral service nodes. should be added to them only.
+resource "aws_security_group" "ephemeral_node" {
+  name        = "ephemeral_node"
+  description = "hosts that host ephemeral services."
+  vpc_id      = module.vpc.vpc_id
+
+  # incoming cassandra clients
+  ingress {
+    from_port   = 9042
+    to_port     = 9042
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming elasticsearch clients.
+  ingress {
+    from_port   = 9200
+    to_port     = 9200
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming minio clients.
+  ingress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming minio clients.
+  ingress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # other cassandra nodes (non-TLS)
+  ingress {
+    from_port   = 7000
+    to_port     = 7000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other cassandra nodes (TLS)
+  ingress {
+    from_port   = 9160
+    to_port     = 9160
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other elasticsearch nodes
+  ingress {
+    from_port   = 9300
+    to_port     = 9300
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other minio nodes
+  ingress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other minio nodes
+  ingress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  tags = {
+    Name = "ephemeral_node"
+  }
+}
+
+# our bastion host
+resource "aws_instance" "bastion-offline" {
```

Comment:
I would suggest to remove all the `aws_instance` blocks from this PR entirely. I think it makes sense to define these on the user level (for us: inside an environment folder), not as a module, depending on requirements for that environment. We can discuss that separately on how best to proceed, and possibly create another PR with another module, if we deem that a good strategy.